pyccel/ast/core.py

```diff
@@ -226,6 +232,41 @@ def handle_precedence(args, my_precedence):
 
     return args
 
+class PyccelBitOperator(Expr, PyccelAstNode):
+    _rank = 0
+    _shape = ()
+    _dtype = NativeInteger()
+
+    def __init__(self, *args):
+        if self.stage == 'syntactic':
+            self._args = handle_precedence(args, self.precedence)
+            return
+        
+        integers = [a for a in args if a.dtype is NativeInteger() or ad.dtype is NativeBool]
+        if integers:
+            self._precision = max(a.precision for a in integers)
```

Comment:
I was thinking the samething, except that I found this way of handling operators is consistent through out the file, take for example `PyccelDiv`, we are not checking if the operands are not strings, I assumed we are just leaving it to the compiler. what you think ?

Reply:
The typo is a good spot. If it is not causing errors then this code is not used in the tests. I think that's because anything that is an Integer is also a Bool.
@saidctb can you confirm this?

Reply:
Actually there are no tests for the case where one of the operands is Bool, I will add them right away

Reply:
It a complicated task to debug python code from another language.
A small and precise error message with its location in the semantic stage is better than a long pyccel error traceback and a complex compiler error that can be hard to read and reference the current python code.
I think we should do the same for other errors instead of let in them to compilers.

Reply:
Readable errors can be printed in the semantics.py file using the errors structure. The error that you removed can serve as an example of how to implement this. If you can also add a test in the errors folder to ensure that it is correctly raised then that's even better

Reply:
no a `NativeInteger()` is not a `NativeBool()`

Reply:
Ah sorry of course. I was thinking of ._is_integer and ._is_boolean. I've got to start thinking more when I answer things at night ^^'